tensorflow/lite/kernels/embedding_lookup_test.cc

```diff
@@ -109,6 +107,39 @@ TEST(EmbeddingLookupOpTest, SimpleTest) {
               })));
 }
 
+TEST(EmbeddingLookupOpTest, VariableShapeTest) {
+  EmbeddingLookupOpModel m({2}, {3, 2, 4});
+  m.SetInput({0, 2});
+  m.Set3DWeightMatrix(
+      [](int i, int j, int k) { return i + j / 10.0f + k / 100.0f; });
+
+  m.Invoke();
+
+  EXPECT_THAT(m.GetOutput(),
+              ElementsAreArray(ArrayFloatNear({
+                  0.00, 0.01, 0.02, 0.03, 0.10, 0.11, 0.12, 0.13,  // Row 0
+                  2.00, 2.01, 2.02, 2.03, 2.10, 2.11, 2.12, 2.13,  // Row 2
+              })));
+}
+
+TEST(EmbeddingLookupOpTest, InvalidShapeTest) {
+  EmbeddingLookupOpModel m({3}, {3, 2, 4});
+  m.SetInput({3, 0, 2});
+  m.Set3DWeightMatrix(
+      [](int i, int j, int k) { return i + j / 10.0f + k / 100.0f; });
+
+  EXPECT_DEATH(m.Invoke(), "Embedding Lookup: index out of bounds.");
+}
+
+TEST(EmbeddingLookupOpTest, NegativeShapeTest) {
```

Comment:
Same comment as "InvalidShapeTest" comment. Can call one "IndexTooHigh" and one "NegativeIndex" to dedup.

We're not testing a negative shape here since all the elements in {3} and {3, 2, 4} are positive.

Reply:
@alanchiao , thanks for the input, i updated the TC as per your comments.

Regards
Amit